Utils/VersionBump/main.swift

```diff
@@ -62,3 +71,50 @@ func extractVersionOnly(_ version: String) -> String {
     guard let indexOfHypen = version.firstIndex(of: "-") else { return version }
     return String(version.prefix(upTo: indexOfHypen))
 }
+
+func verifyVersionInFiles() throws {
+    let expectedVersion = args[2]
+    
+    for file in files {
+        let fileHandler = try open(file)
+        let fileContent: String = fileHandler.read()
+
+        var regexString: String = ""
+        if file.hasSuffix(".podspec") {
+            if file == "./Tests/HybridSDKTest/HybridPod.podspec" {
+                regexString = "s\\.dependency\\s\"Sentry\\/HybridSDK\",\\s\"(?<version>[a-zA-z0-9\\.\\-]+)\""
+            } else {
+                regexString = "\\ss\\.version\\s+=\\s\"(?<version>[a-zA-z0-9\\.\\-]+)\""
+            }
+        } else if file == "./Package.swift" {
+            regexString = "https:\\/\\/github\\.com\\/getsentry\\/sentry-cocoa\\/releases\\/download\\/(?<version>[a-zA-z0-9\\.\\-]+)\\/Sentry"
+        } else if file == "./Sources/Sentry/SentryMeta.m" {
+            regexString = "static NSString \\*versionString = @\"(?<version>[a-zA-z0-9\\.\\-]+)\""
+        }
+        let match = try? Regex(string: regexString, options: [.dotMatchesLineSeparators]).firstMatch(in: fileContent)
+        let version = match?.captures[0] ?? "Version Not Found"
+        if version != expectedVersion {
+            exit(errormessage: "Unexpected version \(version) found for file '\(file)'")
+        } else {
+            print("\(file) validated to have the correct version: \(version)")
+        }
+    }
+    
+    let exactVersion = extractVersionOnly(expectedVersion)
+    for file in restrictFiles {
+        let fileHandler = try open(file)
+        let fileContent: String = fileHandler.read()
+        
+        let marketingRegex = try? Regex(string: "MARKETING_VERSION\\s=\\s(?<version>[a-zA-z0-9\\.\\-]+)", options: [.dotMatchesLineSeparators])
+        let currentProjectRegex = try? Regex(string: "CURRENT_PROJECT_VERSION\\s=\\s(?<version>[a-zA-z0-9\\.\\-]+)", options: [.dotMatchesLineSeparators])
+        let match = marketingRegex?.firstMatch(in: fileContent) ?? currentProjectRegex?.firstMatch(in: fileContent)
+        let version = match?.captures[0] ?? "Version Not Found"
+        if version != exactVersion {
+            exit(errormessage: "Unexpected version \(version) found for file '\(file)'")
```

Comment:
Same comment here, consider iterating through all files so we get a more holistic view of everything that failed.